### PR TITLE
Adds TO client api tests for DS automation

### PR DESCRIPTION
**Pull request by dmohan001c**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Ops
- CI tests

## What is the best way to verify this PR?
Execute all the Integration tests and make sure the tests passed.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by dmohan001c** [comment]
I am not sure why I am getting errors when I am adding these tests. It's working fine in local.

**Comment by ocket8888** [comment]
Error message indicates that `testData.DeliveryServices[0].ProfileName` is nil - are you working with the same tc-fixtures as master?

**Comment by ocket8888** [comment]
yeah, on master I can see:
```
            "profileName": null,
```
So your tests are depending on a specific Delivery Service in the test data having a ProfileName that is safe to dereference, but it just doesn't. It's possible that this is because the fixtures data has changed since the time these tests were written.

**Comment by dmohan001c** [comment]
yes, that's fine. I will update it. But this error is not related to the profile name null value. 
Error Message: Traffic Ops: ERROR: traffic_ops_golang.go:161: 2021-04-14T04:50:58.152866453Z: listen tcp 127.0.0.1:6060: bind: address already in use

**Comment by ocket8888** [comment]
Sounds like your port 6060 is already in use and it's trying to start Traffic Ops listening to that port. `netstat -nlptu | grep 6060` should be able to find the PID/Process name using it. You might need to use `sudo` if the process isn't owned by your user

**Comment by dmohan001c** [comment]
Re-opening pull request to check errors.

File: .github/actions/to-integration-tests/entrypoint.sh [hunk 1]
```diff
@@ -179,13 +181,16 @@ envsubst <"${resources}/riak.json" >riak.conf
...
+go test -test.v --cfg traffic-ops-test.conf || code="$?" && code="$?"
+
+# TODO - Make these logs build artifacts
+# 2>&1 makes terminal output go faster, even though stderr will not contain anything
+echo "------------ TRAFFIC OPS LOGS ------------"
+cd -
+color_and_prefix "${yellow_bg}" 'Traffic Ops' <warning.log 2>&1
+color_and_prefix "${red_bg}" 'Traffic Ops' <error.log 2>&1
+
+exit "$code"
```
Comment (ocket8888) [inline .github/actions/to-integration-tests/entrypoint.sh:196]: Missing newline at EOF

**Review by ocket8888 — APPROVED** [review]
CodeQL problem is a known issue that needs to be addressed under our security policy (it's also nothing; just some "ucontrolled data used in network request" in a testing utility).

**Pull request merged**